### DISCUSSION TITLE
Had anyone luck using llama 3 for function calling?

### DISCUSSION THREAD

**Discussion by WebsheetPlugin** [microsoft/autogen#2549] [General]
I am looking for an cheaper alternative to gpt-4-turbo and was thinking to use llama 3. What is the current state? Does anyone succeeded? Or should I wait for phi? As I imagine Autogen will support it better?

**Comment by ekzhu** [comment]
Take a look at user-defined functions: https://microsoft.github.io/autogen/docs/topics/code-execution/user-defined-functions. You can use it for models that do not support function call. Please let us know how well it works.
